smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/ResourceIdentifierBindingValidator.java

```diff
@@ -66,7 +66,7 @@ private void validateResourceIdentifierTraits(Model model, List<ValidationEvent>
             Map<String, Set<String>> bindings = computePotentialStructureBindings(model, container);
             for (Map.Entry<String, Set<String>> entry : bindings.entrySet()) {
                 if (entry.getValue().size() > 1) {
-                    events.add(error(model.expectShape(container), String.format(
+                    events.add(warning(model.expectShape(container), String.format(
```

Comment:
We should not allow explicit binding conflicts. Warning should only warn when explicit bindings are used and there's a viable implicit binding of the same name that isn't used (since it's confusing). Explicit binding collisions should be error.

e.g., this must be an error:

```
structure Foo {
    @required
    @resourceIdentifier("foo")
    a: String

    @required
    @resourceIdentifier("foo")
    b: String
}

```